core, eth, miner, xeth: clean up tx/receipt db accessors

karalabe opened the pull request:
*This PR will need to be rebased when https://github.com/ethereum/go-ethereum/pull/1917 is merged. Until then it's on hold to prevent further merge conflicts.*

Additionally, it cleans up the transaction and receipt database accessors:
 * [x] Unify naming convention between chain and txn database ops.
 * [x] Unite the db util files into a single source (`chain_util`, `transaction_util` -> `database_util`)
 * [x] Add tests for receipt, transaction and block-receipts storage methods.
 * [x] Refactor `xeth` to use the db accessor methods, and not poke in the db directly.

robotally commented:
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Oct 22 16:32:46 UTC 2015_

Gustav-Simonsson commented:
:+1:

karalabe commented:
@Gustav-Simonsson Btw, just rebased this, fixed a tiny merge conflict and force pushed. Might want to check skim through it to ensure everything's still fine :)

@obscuren Rebased on top of tip, PTAL.
